### PR TITLE
improve persist car

### PR DESCRIPTION
**Pull request by TheGreenBandit**
thanks Lord Griffin and tupoya for helping implement this

Original post:
broken somehow, needs fix before implementing 

issues:
1,	'byte': ambiguous symbol (compiling source file src\services\vehicle\persist_car_service.cpp)	BigBaseV2	C:\Users\TGB\YimMenuAdditions\BigBaseV2\src\util	C:\Users\TGB\YimMenuAdditions\BigBaseV2\src\util\world_model.hpp	line 10

2, 	'initializing': cannot convert from 'const Vector3' to 'Hash'	BigBaseV2	C:\Users\TGB\YimMenuAdditions\BigBaseV2\src\services\vehicle	C:\Users\TGB\YimMenuAdditions\BigBaseV2\src\services\vehicle\persist_car_service.cpp	line 275

### CODE REVIEW

**Comment by TheGreenBandit** [comment]
once issues are fixed, this will improve persist car in allowing collision, visibility and invincibility to be edited on attachments and base models

**Comment by TheGreenBandit** [comment]
all that needs adding is invincibility, thanks lord griffin for doing this

**Comment by TheGreenBandit** [comment]
Btw with this u would have to resave all ur jsons, loading them from and older version of yim and then saving them on the newer version

File: BigBaseV2/src/core/data/model_attachment.hpp [hunk 1]
```diff
@@ -6,22 +6,34 @@ namespace big
 {
 	struct model_attachment
 	{
+		int model_type;
```
Comment (xiaoxiao921) [inline BigBaseV2/src/core/data/model_attachment.hpp]: why is this even here? just duplicate of `model_hash` ?

File: BigBaseV2/src/core/data/model_attachment.hpp [hunk 2]
```diff
@@ -6,22 +6,34 @@ namespace big
...
+		bool visible;
+		bool invincible;
+
 	};
+	void to_json(nlohmann::json& j, const model_attachment& attachment) {
 
-	static void to_json(nlohmann::json& j, const model_attachment& attachment)
-	{
-		j = nlohmann::json{ {"model_hash", attachment.model_hash},
+		j = nlohmann::json{ {"model_type", attachment.model_type} ,{"model_hash", attachment.model_hash},
```
Comment (xiaoxiao921) [inline BigBaseV2/src/core/data/model_attachment.hpp]: why is this even here? just duplicate of model_hash ?

File: BigBaseV2/src/core/data/model_attachment.hpp [hunk 3]
```diff
@@ -6,22 +6,34 @@ namespace big
...
 							{"position_x", attachment.position.x}, {"position_y", attachment.position.y}, {"position_z", attachment.position.z},
-							{"rotation_x", attachment.rotation.x}, {"rotation_y", attachment.rotation.y}, {"rotation_z", attachment.rotation.z} };
-	}
+							{"rotation_x", attachment.rotation.x}, {"rotation_y", attachment.rotation.y}, {"rotation_z", attachment.rotation.z},
+			{"has_collsion", attachment.collision},
```
Comment (xiaoxiao921) [inline BigBaseV2/src/core/data/model_attachment.hpp]: Fix spelling mistakes and make the json name and the name in the struct match

File: BigBaseV2/src/core/data/model_attachment.hpp [hunk 4]
```diff
@@ -6,22 +6,34 @@ namespace big
...
-	{
+		};
+	}
+//thank you Lord Griffin of Nenyoo for sharing how
+	void from_json(const nlohmann::json& j, model_attachment& attachment) {
+		j.at("model_type").get_to(attachment.model_type);
 		j.at("model_hash").get_to(attachment.model_hash);
 		j.at("position_x").get_to(attachment.position.x); j.at("position_y").get_to(attachment.position.y); j.at("position_z").get_to(attachment.position.z);
 		j.at("rotation_x").get_to(attachment.rotation.x); j.at("rotation_y").get_to(attachment.rotation.y); j.at("rotation_z").get_to(attachment.rotation.z);
+		j.at("has_collsion").get_to(attachment.collision);
```
Comment (xiaoxiao921) [inline BigBaseV2/src/core/data/model_attachment.hpp]: Fix spelling mistakes and make the json name and the name in the struct match

File: BigBaseV2/src/services/vehicle/persist_car_service.hpp [hunk 5]
```diff
@@ -20,6 +20,8 @@ namespace big
 		static constexpr auto model_attachments_key = "model_attachments";
 
 		static constexpr auto vehicle_attachments_key = "vehicle_attachments";
+		static constexpr auto is_visible_key = "is_visible";
```
Comment (xiaoxiao921) [inline BigBaseV2/src/services/vehicle/persist_car_service.hpp:23]: if you add the keys then atleast use them in the actual code

File: BigBaseV2/src/services/vehicle/persist_car_service.hpp [hunk 6]
```diff
@@ -78,3 +80,9 @@ namespace big
 		static big::folder check_vehicle_folder();
 	};
 }
+using namespace std;
```
Comment (xiaoxiao921) [inline BigBaseV2/src/services/vehicle/persist_car_service.hpp]: No, why is that here?

File: BigBaseV2/src/services/vehicle/persist_car_service.hpp [hunk 7]
```diff
@@ -78,3 +80,9 @@ namespace big
...
+
+namespace JSON
```
Comment (xiaoxiao921) [inline BigBaseV2/src/services/vehicle/persist_car_service.hpp]: why a whole namespace for this? remove that and just put that as static member method like the rest

File: README.md [hunk 8]
```diff
@@ -38,12 +38,12 @@ Premake5 is used to generate our project files, if you haven't used it before we
 
 Clone the repository including submodules:
 ```bash
-git clone https://github.com/YimMenu/YimMenu.git --recursive || echo "You don't have git installed, install it from https://git-scm.com/download/win"
+git clone https://github.com/TheGreenBandit/YimMenuAdditions.git --recursive || echo "You don't have git installed, install it from https://git-scm.com/download/win"
```
Comment (xiaoxiao921) [inline README.md]: Revert that README change

File: README.md [hunk 9]
```diff
@@ -38,12 +38,12 @@ Premake5 is used to generate our project files, if you haven't used it before we
...
 ```
 
 Go into the directory you just cloned:
 ```bash
-cd YimMenu
+cd YimMenuAdditons
```
Comment (xiaoxiao921) [inline README.md]: Same

File: BigBaseV2/src/services/vehicle/persist_car_service.cpp [hunk 10]
```diff
@@ -389,6 +401,10 @@ namespace big
 		VEHICLE::GET_VEHICLE_EXTRA_COLOURS(vehicle, &pearlescent_color, &wheel_color);
 
 		vehicle_json[pearlescent_color_key] = pearlescent_color;
+		int visible = ENTITY::IS_ENTITY_VISIBLE(vehicle);
```
Comment (xiaoxiao921) [inline BigBaseV2/src/services/vehicle/persist_car_service.cpp]: Cast to a proper type, `bool`

File: BigBaseV2/src/services/vehicle/persist_car_service.cpp [hunk 11]
```diff
@@ -145,6 +146,17 @@ namespace big
 			std::vector<int> primary_custom_color = vehicle_json[custom_primary_color_key];
 			VEHICLE::SET_VEHICLE_CUSTOM_PRIMARY_COLOUR(vehicle, primary_custom_color[0], primary_custom_color[1], primary_custom_color[2]);
 		}
+		if (!vehicle_json[is_visible_key].is_null())
+		{
+			int visible = vehicle_json[is_visible_key];
```
Comment (xiaoxiao921) [inline BigBaseV2/src/services/vehicle/persist_car_service.cpp]: Cast to a proper type, `bool`

**Review by xiaoxiao921 — DISMISSED** [review]

**Comment by TheGreenBandit** [comment]
Xiao ill give u write access to repository, i cant get on my pc at all today really because of football practice and school

**Comment by TheGreenBandit** [comment]
Tupoya has fixed the issue, it should be ready for merge

**Comment by TheGreenBandit** [comment]
What happend to the discord btw?

**Comment by tupoy-ya** [comment]
> What happend to the discord btw?

https://github.com/YimMenu/YimMenu/discussions/361#discussioncomment-3734636

It's matrix only now.

**Comment by Yimura** [comment]
Check [this discussion comment](https://github.com/YimMenu/YimMenu/discussions/361#discussioncomment-3732080)

**Comment by TheGreenBandit** [comment]
XiaoXiao remove ur requested changes as everything is resolved and ready to merge

**Comment by GeopJr** [comment]
It doesn't matter, someone has to approve this (probably Yimura) before it gets merged.

**Comment by TheGreenBandit** [comment]
Alr, this will be a nice addition to the menu once approved

**Review by Yimura — DISMISSED** [review]

**Comment by GeopJr** [comment]
Hey @TheGreenBandit, does "Lord Griffin" have a GitHub account or an email so we can set them as a co-author (instead of having them as a comment)?

**Comment by TheGreenBandit** [comment]
His github: https://github.com/navmodder

His discord: Peter Griffin#3056

**Review by GeopJr — APPROVED** [review]

**Pull request merged**